woocommerce/woocommerce-android
# Issue/12750 blaze survey

Closes: #12750 


⚠️ Requires [these FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3105) changes to work

### Description

Adds a new feedback request UI component in Blaze campaign creation success bottom sheet. Upon interacting with either 👍🏼 or 👎🏼 from the feedback component, a survey webview will be displayed

### Steps to reproduce


### Testing information
Apply this patch to disable the Objective feature flag and be able to create a campaign without setting a mandatory objective value:

```diff
Subject: [PATCH] Disable Objective feature flag in debug
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt	(revision c24bfbad0ba216ce633a385bb80b89cf3712bf2d)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt	(date 1727972939935)
@@ -25,11 +25,11 @@
                 PackageUtils.isDebugBuild() || context != null && PackageUtils.isBetaBuild(context)
             }
 
+            OBJECTIVE_SECTION -> false
             WC_SHIPPING_BANNER,
             BETTER_CUSTOMER_SEARCH_M2,
             ORDER_CREATION_AUTO_TAX_RATE,
-            REVAMP_WOO_SHIPPING,
-            OBJECTIVE_SECTION -> PackageUtils.isDebugBuild()
+            REVAMP_WOO_SHIPPING -> PackageUtils.isDebugBuild()
 
             NEW_SHIPPING_SUPPORT,
             INBOX,

```

1. Create a Blaze campaign
2. In the success bottom sheet click on any of the 👍🏼 or 👎🏼
3. Check the survey is loaded 
4. Check in the Android logcat the following log: 
`🔵 Tracked: blaze_campaign_creation_feedback, Properties: {"satisfied":true...`

### The tests that have been performed

The above ☝🏼. Created a Blaze campaign and opened the survey form

### Images/gif


https://github.com/user-attachments/assets/c78eec3f-bb48-4e6a-9009-5ecb7004c4df


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
